test/e2e/upgrade_test.go

```diff
@@ -74,6 +79,7 @@ func upgradeTestSuite(t *testing.T) {
 	require.NoError(t, err, "wrong number of ConfigMap controller nodes found")
 
 	t.Run("Operator version upgrade", testUpgradeVersion)
+	t.Run("Dependent service reconcilation", testCtx.testDependentServiceChanges)
```

Comment:
Why is this test in the upgrade test suite?

Reply:
Somewhat arbitrary, but I figured this specific situation is more likely in an upgrade scenario. I'm fine moving it to elsewhere if you think it should be run on all platforms.

Reply:
I'm fine with having this as part of the upgrade tests as long as it doesn't greatly increase the time the suite takes to complete. No need to run on all platforms IMO

Reply:
I have the same concern with the length of the upgrade test increasing. It is the prime candidate for overrunning the 4hr time out.